backend/controller/observability/observability.go

```diff
@@ -0,0 +1,195 @@
+package observability
+
+import (
+	"context"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"time"
+
+	"github.com/alecthomas/types/optional"
+
+	"github.com/TBD54566975/ftl/backend/controller/dal"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+type metricAttributeBuilders struct {
+	moduleName      func(name string) attribute.KeyValue
+	featureName     func(name string) attribute.KeyValue
+	destinationVerb func(name string) attribute.KeyValue
+}
+
+type callMetrics struct {
+	meter    metric.Meter
+	requests metric.Int64Counter
+	failures metric.Int64Counter
+	active   metric.Int64UpDownCounter
+	latency  metric.Int64Histogram
+}
+
+type fsmMetrics struct {
+	meter       metric.Meter
+	active      metric.Int64UpDownCounter
+	transitions metric.Int64Counter
+}
+
+type observableMetrics struct {
+	attributes metricAttributeBuilders
+	calls      *callMetrics
+	fsm        *fsmMetrics
+}
```

Comment:
Quite a bit of this is duplicative as well, so we could just get rid of the structs. We need to instantiate the metrics below anyway, so this is just another site we'd need to add a line whenever we need to add a new metric.

Reply:
yep good call, I'll incorporate that input as I break out the metrics into individual feature based metrics (e.g. observability/calls.go, etc)